src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -140,24 +139,54 @@ protected override void ProcessRecord()
 
                 if (_jschema != null)
                 {
-                    var errorMessages = _jschema.Validate(parsedJson);
-                    if (errorMessages != null && errorMessages.Count != 0)
+                    try
                     {
-                        result = false;
-
-                        Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema);
-
-                        foreach (var message in errorMessages)
+                        var validationResults = _jschema.Validate(parsedJson, new ValidationOptions { OutputFormat = OutputFormat.Basic });
+                        result = validationResults.IsValid;
+                        if (!result)
                         {
-                            ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema", ErrorCategory.InvalidData, null);
-                            errorRecord.ErrorDetails = new ErrorDetails(message.ToString());
-                            WriteError(errorRecord);
+                            Exception exception = new(TestJsonCmdletStrings.InvalidJsonAgainstSchema);
+
+                            if (validationResults.Message != null)
+                            {
+                                ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema",
+                                    ErrorCategory.InvalidData, null);
+                                var message = $"{validationResults.Message} at {validationResults.InstanceLocation}";
+                                errorRecord.ErrorDetails = new ErrorDetails(message);
+                                WriteError(errorRecord);
+                            }
+
+                            if (validationResults.HasNestedResults)
+                            {
+                                foreach (var nestedResult in validationResults.NestedResults)
+                                {
+                                    if (nestedResult.Message == null)
+                                    {
+                                        continue;
+                                    }
+
+                                    ErrorRecord errorRecord = new(exception, "InvalidJsonAgainstSchema", ErrorCategory.InvalidData, null);
+                                    var message = $"{nestedResult.Message} at {nestedResult.InstanceLocation}";
+                                    errorRecord.ErrorDetails = new ErrorDetails(message);
+                                    WriteError(errorRecord);
+                                }
+                            }
                         }
                     }
+                    // A JsonException that occurs during validation is the result of a failure to deserialize
+                    // a referenced schema (through $ref or $dynamicRef).  References are resolved at validation
+                    // time and deserialized on-demand.
+                    catch (JsonException jsonExc)
```

Comment:
Please move the comment in catch block to avoid style warning.

Also we should catch all exceptions here as commented above.
"So we should catch all exceptions from HttpClient and File.ReadAllText()."

Reply:
There are a _lot_ of potential exceptions that come from those.  I just wrapped the fetch code in a try/catch and had it wrap the exception in a new one.  Then I could just catch that exception below.